### PR TITLE
fix(theme): make select styles consistent with input

### LINKED ISSUE #3876: [BUG] - Select component and the Input component style inconsistently
> * Flat: Currently the bg-color becomes darker on hover

I'm sorry, but I personally think there is an issue with this change, as it causes the Select component and the Input component to behave inconsistently.

recording

_Originally posted by @pompurin404 in https://github.com/nextui-org/nextui/issues/3663#issuecomment-2380399790_

### PR DESCRIPTION
**Pull request by macci001**
Closes #3876 

#### 📝 Description

The changes in this PR makes the select and input styles consistent.

#### ⛳️ Current behaviour (updates)

https://github.com/user-attachments/assets/12d63c9c-22fb-4402-9d2d-f2894b918b73

#### 🚀 New behaviour

* Flat Variant
   * Changes are made to make the hover color consistent

https://github.com/user-attachments/assets/ec2dfff4-ec5c-477d-af98-ff96d817c2fb

https://github.com/user-attachments/assets/ec67fdff-9cd3-4ca8-bfed-65ef1fa85ff2

* Faded
   * Removing the bg-color change on hover
   * BorderColor added in case of focus and popover-open 

https://github.com/user-attachments/assets/f899f35f-d46f-4a52-80e1-70900b812e88

* Bordered
  *  No styling change, just formatting change to remove the repeated style
* Underlined
  *  No changes.

## Summary by CodeRabbit

- **New Features**
	- Updated versioning for multiple `@nextui-org` packages, indicating potential new features and improvements.
- **Style**
	- Enhanced visual feedback for input and select components with updated hover and focus states.
	- Adjusted background and border colors for various input and select variants to improve user interaction.
- **Documentation**
	- Exported new types and variables for input and select components, improving accessibility for developers.

### CODE REVIEW

File: packages/core/theme/src/components/select.ts [hunk 1]
```diff
@@ -224,7 +224,11 @@ const select = tv({
       color: "default",
       class: {
         value: "group-data-[has-value=true]:text-default-foreground",
-        trigger: ["bg-default-100", "data-[hover=true]:bg-default-50"],
+        trigger: [
+          "bg-default-100",
+          "data-[hover=true]:bg-default-200",
+          "group-data-[focus=true]:bg-default-50",
+        ],
```
[the remark this answers is not in the document]
Comment (macci001) [reply packages/core/theme/src/components/select.ts:231]: Resolving as this just formatting change in the lines mentioned.

File: packages/core/theme/src/components/select.ts [hunk 2]
```diff
@@ -234,7 +238,7 @@ const select = tv({
         trigger: [
           "bg-primary-100",
           "text-primary",
-          "data-[hover=true]:bg-primary-50",
+          "data-[hover=true]:bg-primary-200",
```
[the remark this answers is not in the document]
Comment (macci001) [reply packages/core/theme/src/components/select.ts:241]: incorrect review comment

File: packages/core/theme/src/components/select.ts [hunk 3]
```diff
@@ -248,7 +252,7 @@ const select = tv({
         trigger: [
           "bg-secondary-100",
           "text-secondary",
-          "data-[hover=true]:bg-secondary-50",
+          "data-[hover=true]:bg-secondary-200",
```
[the remark this answers is not in the document]
Comment (macci001) [reply packages/core/theme/src/components/select.ts:255]: incorrect review comment.

**Review by wingkwong — CHANGES_REQUESTED** [review]
Except the default one, all other colors look different compared with those in input. Currently the pr changes make the contrast lower in dark mode. You may compare Colors section in Select and Input.

**Review by dacitto — APPROVED** [review]

**Comment by jrgarciadev** [comment]
Hey @macci001 thanks for this contribution, let's do the opposite, let's make the input look like the select styles -- the select hover colors look better

**Comment by macci001** [comment]
@jrgarciadev sure, will do!

**Comment by jrgarciadev** [comment]
Thank you! @macci001

**Pull request merged**